File: Magestorm2/Server/MagestormServer/src/DisciplineData.java
import java.util.ArrayList;

public class DisciplineData {
    private final byte _id, _availabilityMask, _statCode;

    public DisciplineData(byte disciplineID, byte[] attrib){
        _id = disciplineID;
        _availabilityMask = attrib[0];
        ParseMask();
        _statCode = attrib[1];
    }

    public byte GetDisciplineID(){
        return _id;
    }

    public byte GetAvailabilityMask(){
        return _availabilityMask;
    }

    public byte GetStatCode(){
        return _statCode;
    }
    private void ParseMask(){
        ArrayList<Byte> classCodes = CharacterClassManager.GetClassCodes();
        boolean[] bools = ByteUtils.BytesToBooleanArray(new byte[]{_availabilityMask});
        for(byte classCode : classCodes){
            if(bools[classCode]){
                CharacterClassManager.AddDisciplineToClass(classCode, this);
            }
        }
    }
}

File: Magestorm2/Assets/Utility/Shared/DisciplineData.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

public class DisciplineData
{
    private byte _id;
    private short _nameRef;
    private byte _availabilityMask;
    private byte _statCode;

    private List<SpellData> _spellsOfDiscipline;

    public DisciplineData(string[] fields, string contents)
    {
        _spellsOfDiscipline = new List<SpellData>();
        string[] split = contents.Split("<br>");
        string fieldID, fieldValue;
        for (int i = 0; i < fields.Length; i++)
        {
            fieldID = fields[i];
            fieldValue = split[i + 1];
            switch (fieldID)
            {
                case DisciplineAttributes.ID:
                    _id = byte.Parse(fieldValue);
                    break;
                case DisciplineAttributes.NAMEREFERENCE:
                    _nameRef = short.Parse(fieldValue);
                    break;
                case DisciplineAttributes.AVAILABILITYMASK:
                    _availabilityMask = byte.Parse(fieldValue);
                    ParseMask();
                    break;
            }
        }
    }
    private void ParseMask()
    {
        BitArray ba = ByteUtils.ByteToBoolArray(_availabilityMask);
        List<byte> classCodes = CharacterClassManager.CharacterClassCodes;
        foreach(byte classCode in classCodes)
        {
            if (ba[classCode])
            {
                CharacterClassManager.AssociateDisciplineToClass(classCode, this);
            }
        }
    }
    public void AddSpellToDiscipline(SpellData spell)
    {
        _spellsOfDiscipline.Add(spell);
    }
    public List<SpellData> SpellsOfDiscipline
    {
        get { return _spellsOfDiscipline; }
    }
    public byte DisciplineID
    {
        get { return _id; }
    }
    public string DisciplineName
    {
        get
        {
            return Language.GetBaseString(_nameRef);
        }
    }
}

public static class DisciplineAttributes
{
    public const string ID = "id";
    public const string NAME = "name";
    public const string AVAILABILITYMASK = "availabilitymask";
    public const string STATCODE = "statcode";
    public const string NAMEREFERENCE = "nameref";
}